dateutil/test/test_rrule.py

```diff
@@ -2853,6 +2853,74 @@ def testStrSetExDate(self):
                           datetime(1997, 9, 9, 9, 0),
                           datetime(1997, 9, 16, 9, 0)])
 
+    def testStrSetExDateMultiple(self):
+        rrstr = ("DTSTART:19970902T090000\n"
+                 "RRULE:FREQ=YEARLY;COUNT=6;BYDAY=TU,TH\n"
+                 "EXDATE:19970904T090000,19970911T090000,19970918T090000\n")
+
+        rr = rrulestr(rrstr)
+        assert list(rr) == [datetime(1997, 9, 2, 9, 0),
+                            datetime(1997, 9, 9, 9, 0),
+                            datetime(1997, 9, 16, 9, 0)]
+
+    def testStrSetExDateWithTZID(self):
+        BXL = tz.gettz('Europe/Brussels')
+        rr = rrulestr("DTSTART;TZID=Europe/Brussels:19970902T090000\n"
+                      "RRULE:FREQ=YEARLY;COUNT=6;BYDAY=TU,TH\n"
+                      "EXDATE;TZID=Europe/Brussels:19970904T090000\n"
+                      "EXDATE;TZID=Europe/Brussels:19970911T090000\n"
+                      "EXDATE;TZID=Europe/Brussels:19970918T090000\n")
+
+        assert list(rr) == [datetime(1997, 9, 2, 9, 0, tzinfo=BXL),
+                            datetime(1997, 9, 9, 9, 0, tzinfo=BXL),
+                            datetime(1997, 9, 16, 9, 0, tzinfo=BXL)]
+
+    def testStrSetExDateValueDateTimeNoTZID(self):
+        rrstr = '\n'.join([
+            "DTSTART:19970902T090000",
+            "RRULE:FREQ=YEARLY;COUNT=4;BYDAY=TU,TH",
+            "EXDATE;VALUE=DATE-TIME:19970902T090000",
+            "EXDATE;VALUE=DATE-TIME:19970909T090000",
+        ])
+
+        rr = rrulestr(rrstr)
+        assert list(rr) == [datetime(1997, 9, 4, 9), datetime(1997, 9, 11, 9)]
+
+    def testStrSetExDateValueMixDateTimeNoTZID(self):
+        rrstr = '\n'.join([
+            "DTSTART:19970902T090000",
+            "RRULE:FREQ=YEARLY;COUNT=4;BYDAY=TU,TH",
+            "EXDATE;VALUE=DATE-TIME:19970902T090000",
+            "EXDATE:19970909T090000",
+        ])
+
+        rr = rrulestr(rrstr)
+        assert list(rr) == [datetime(1997, 9, 4, 9), datetime(1997, 9, 11, 9)]
+
+    def testStrSetExDateValueDateTimeWithTZID(self):
+        BXL = tz.gettz('Europe/Brussels')
+        rrstr = '\n'.join([
+            "DTSTART;VALUE=DATE-TIME;TZID=Europe/Brussels:19970902T090000",
+            "RRULE:FREQ=YEARLY;COUNT=4;BYDAY=TU,TH",
+            "EXDATE;VALUE=DATE-TIME;TZID=Europe/Brussels:19970902T090000",
+            "EXDATE;VALUE=DATE-TIME;TZID=Europe/Brussels:19970909T090000",
+        ])
+
+        rr = rrulestr(rrstr)
+        assert list(rr) == [datetime(1997, 9, 4, 9, tzinfo=BXL),
+                            datetime(1997, 9, 11, 9, tzinfo=BXL)]
+
+    def testStrSetExDateValueDate(self):
+        rrstr = '\n'.join([
+            "DTSTART;VALUE=DATE:19970902",
+            "RRULE:FREQ=YEARLY;COUNT=4;BYDAY=TU,TH",
+            "EXDATE;VALUE=DATE:19970902",
+            "EXDATE;VALUE=DATE:19970909",
```

Comment:
This test makes me realize that the code handling TZID shouldn't be triggered when it's a VALUE=DATE, should it?

Reply:
I'm not actually sure. It's certainly the case that the `datetime.date` has no way to represent a time zone, but what `dateutil.rrule` does is to parse `DATE` values as "datetime at midnight", which means they *can* have a time zone.

It depends on what the spec says, I suppose. [The grammar for EXDATE](https://tools.ietf.org/html/rfc5545#section-3.8.5.1) seems to indicate it's syntactically valid to have a `TZID` for `VALUE=DATE`, but maybe it's forbidden at a semantic level elsewhere in the spec.